test/server/cards/01_SOR/units/K2SOCassiansCounterpart.spec.ts

```diff
@@ -0,0 +1,53 @@
+describe('K-2SO', function() {
+    integration(function(contextRef) {
+        describe('K-2SO\'s When Defeated ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['k2so#cassians-counterpart'],
+                    },
+                    player2: {
+                        hand: ['wampa'],
+                        groundArena: ['krayt-dragon'],
+                    }
+                });
+            });
+
+            it('should either deal 3 damage to the opponent\'s base or the opponent discards a card from their hand', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.k2so);
+                context.player1.clickCard(context.kraytDragon);
+                expect(context.player1).toHaveExactPromptButtons(['Deal 3 damage to opponent\'s base', 'The opponent discards a card']);
+                expect(context.player1).not.toHavePassAbilityButton();
+
+                context.player1.clickPrompt('The opponent discards a card');
+                expect(context.player2.handSize).toBe(0);
+                expect(context.wampa).toBeInLocation('discard');
+
+                context.player1.moveCard(context.k2so, 'ground arena');
+                context.player2.clickCard(context.kraytDragon);
+                context.player2.clickCard(context.k2so);
+                expect(context.player1).toHaveExactPromptButtons(['Deal 3 damage to opponent\'s base', 'The opponent discards a card']);
+                expect(context.player1).not.toHavePassAbilityButton();
+
+                context.player1.clickPrompt('The opponent discards a card');
+                expect(context.player2.handSize).toBe(0);
+
+                context.player1.moveCard(context.k2so, 'ground arena');
+                context.moveToNextActionPhase();
+
+                context.player1.clickCard(context.k2so);
+                context.player1.clickCard(context.kraytDragon);
+                expect(context.p2Base.damage).toBe(2);
+
+                expect(context.player1).toHaveExactPromptButtons(['Deal 3 damage to opponent\'s base', 'The opponent discards a card']);
+                expect(context.player1).not.toHavePassAbilityButton();
+
+                context.player1.clickPrompt('Deal 3 damage to opponent\'s base');
+                expect(context.p2Base.damage).toBe(5);
```

Comment:
I think you should add a test with no hand for opponent to see if they is not prompt to choose between both choice

Reply:
Lines 29-36 are covering the case where the opponent doesn't have any card. I'm not sure the option should be auto selected though, wouldn't be possible to pick "discard a card" even if the opponent has no card? I don't think there is a requirement to change the game state here.

Reply:
Yeah the option should still be available